include/CoreGraphics/CoreGraphicsExport.h

```diff
@@ -66,5 +67,94 @@ typedef struct __CGPath* CGMutablePathRef;
 // typedef struct __Foo* FooRef;
 // indiciating backing code need some clean up
 typedef void* CGDataProviderRef;
+
+#if defined(__clang__)
 typedef id CGFontRef;
-typedef id CGPatternRef;
\ No newline at end of file
+typedef id CGPatternRef;
+#endif
+
+typedef enum {
+    kCGImageAlphaNone,
+    kCGImageAlphaPremultipliedLast,
+    kCGImageAlphaPremultipliedFirst,
+    kCGImageAlphaLast,
+    kCGImageAlphaFirst,
+    kCGImageAlphaNoneSkipLast,
+    kCGImageAlphaNoneSkipFirst,
+    kCGImageAlphaOnly
+} CGImageAlphaInfo;
+enum {
+    kCGBitmapAlphaInfoMask = 0x1F,
+    kCGBitmapFloatComponents = (1 << 8),
+    kCGBitmapByteOrderMask = 0x7000,
+    kCGBitmapByteOrderDefault = (0 << 12),
+    kCGBitmapByteOrder16Little = (1 << 12),
+    kCGBitmapByteOrder32Little = (2 << 12),
+    kCGBitmapByteOrder16Big = (3 << 12),
+    kCGBitmapByteOrder32Big = (4 << 12)
+};
+typedef int CGBitmapInfo;
+
+typedef enum {
+    kCGRenderingIntentDefault,
+    kCGRenderingIntentAbsoluteColorimetric,
+    kCGRenderingIntentRelativeColorimetric,
+    kCGRenderingIntentSaturation,
+    kCGRenderingIntentPerceptual,
+} CGColorRenderingIntent;
+
+typedef enum {
+    kCGColorSpaceModelUnknown = -1,
+    kCGColorSpaceModelMonochrome,
+    kCGColorSpaceModelRGB,
+    kCGColorSpaceModelCMYK,
+    kCGColorSpaceModelLab,
+    kCGColorSpaceModelDeviceN,
+    kCGColorSpaceModelIndexed,
+    kCGColorSpaceModelPattern,
+} CGColorSpaceModel;
+
+typedef enum {
+    _Color565 = 0,
+    _ColorARGB,
+    _ColorABGR,
+    _ColorBGRX,
+    _ColorXBGR,
+    _ColorGrayscale,
+    _ColorBGR,
+    _ColorA8,
+    _ColorIndexed,
+    _ColorMax,
+    _ColorRGBA = _ColorABGR
+} surfaceFormat;
+
+typedef struct {
+    CGColorSpaceModel colorSpaceModel;
+    CGBitmapInfo bitmapInfo;
+    unsigned int bitsPerComponent;
+    unsigned int bytesPerPixel;
+} __CGFormatProperties;
+
+static const __CGFormatProperties c_FormatTable[_ColorMax] = {
+    { kCGColorSpaceModelRGB, (kCGImageAlphaNone | kCGBitmapByteOrderDefault), 5, 2 }, //_Color565,
+    { kCGColorSpaceModelRGB, (kCGImageAlphaFirst | kCGBitmapByteOrderDefault), 8, 4 }, //_ColorARGB,
+    { kCGColorSpaceModelRGB, (kCGImageAlphaLast | kCGBitmapByteOrder32Little), 8, 4 }, //_ColorABGR,
+    { kCGColorSpaceModelRGB, (kCGImageAlphaNoneSkipFirst | kCGBitmapByteOrderDefault), 8, 4 }, //_ColorBGRX,
+    { kCGColorSpaceModelRGB, (kCGImageAlphaNoneSkipLast | kCGBitmapByteOrder32Little), 8, 4 }, //_ColorXBGR,
+    { kCGColorSpaceModelMonochrome, (kCGImageAlphaNone | kCGBitmapByteOrderDefault), 8, 1 }, //_ColorGrayscale,
+    { kCGColorSpaceModelRGB, (kCGImageAlphaNone | kCGBitmapByteOrderDefault), 8, 3 }, //_ColorBGR,
+    { kCGColorSpaceModelPattern, (kCGImageAlphaOnly | kCGBitmapByteOrderDefault), 8, 1 }, //_ColorA8,
+    { kCGColorSpaceModelIndexed, (kCGImageAlphaNone | kCGBitmapByteOrderDefault), 8, 2 }, //_ColorIndexed,
+};
+
+struct __CGSurfaceInfo {
+    size_t width;
+    size_t height;
+    size_t bitsPerComponent;
```

Comment:
I agree that the type should be consistent. Will make them all unsigned ints for now since the imageData backings use that type as well and it is highly unlikely we will have dimensions or individual pixel bit/byte sizes in the millions :)